### PR TITLE
Task6 redux

### PR DESCRIPTION
**Pull request by inductio**

### CODE REVIEW

File: src/components/Header/AddMovie.js [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-                </Modal>
-            }
         </>
     )
 };
 
-export default AddMovie;
+const mapStateToProps = (state) => {
+    return { showModal: state.showModal}
+};
```
Comment (lotrien) [inline src/components/Header/AddMovie.js]: if `showModal` is a function:
- it should be a part of `mapDispatchToProps` and be dispatched properly
- functions should not be stored within global store in redux

This is a very crucial moment

File: src/components/Main/MoviesGrid.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
-export default MoviesGrid;
+    const movies = props.activeMovies && props.activeMovies.length ? props.activeMovies : props.movies;
+    return (
+        <div className="movies__grid">
+            {movies.map((movie) => <MovieCard key={movie.id} movie={movie}/>)}
+        </div>
+    )
+
+};
+
```
Comment (lotrien) [inline src/components/Main/MoviesGrid.js]: how about this 
```
useEffect(() => {
  props.fetchMovies();
}, [])
```
thus we will trigger GET movies

and then 
```
if (!movies.length) {
   return;
}
return (<div>RENDER MOVIES HERE</div)
```

File: src/components/movie/DeleteMovieForm.js [hunk 3]
```diff
@@ -19,4 +21,11 @@ const DeleteMovieForm = () => {
     return <Form config={deleteFormConfig}/>
 };
 
-export default DeleteMovieForm;
+const mapState = (state) => {
+    return {
+        deleteMovieRequest: state.deleteMovieRequest,
```
Comment (lotrien) [inline src/components/movie/DeleteMovieForm.js]: if it's a function it should be moved to `mapDispatchToProps` and should NOT be a part of store data

File: src/components/movie/EditMovieForm.js [hunk 4]
```diff
@@ -11,21 +18,32 @@ const EditMovieForm = ({movie}) => {
...
-            <MovieFormFields movie={movie}/>
+            <MovieFormFields movie={props.movie}/>
         </Form>
     )
 };
 
-export default EditMovieForm;
+const mapState = (state) => {
+    return {
+        editMovieRequest: state.editMovieRequest,
```
Comment (lotrien) [inline src/components/movie/EditMovieForm.js]: if it's a function it should be moved to `mapDispatchToProps` and should NOT be a part of store data

File: src/components/movie/MovieCardMenu.js [hunk 5]
```diff
@@ -1,48 +1,37 @@
...
     )
 };
 
-export default MovieCardMenu;
+const mapStateToProps = (state, ownProps) => {
+    return { showModal: state.showModal, ...ownProps}
+};
+
+export default connect(mapStateToProps, {
+    showModal
```
Comment (lotrien) [inline src/components/movie/MovieCardMenu.js]: is it a name collision ?

File: src/index.js [hunk 6]
```diff
@@ -1,11 +1,23 @@
...
 import App from './components/App';
 import ErrorBoundary from './components/common/ErrorBoundary';
+import reducers from './reducers';
+import { createStore, applyMiddleware, compose } from 'redux';
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+
+const store = createStore(reducers, composeEnhancers(
+    applyMiddleware(thunk)
+));
```
Comment (lotrien) [inline src/index.js:14]: it's a common practice to init store in separate file and import it here.

One more thing is that `window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__` is going to be a part of `prod` bundle, but do we need it there/. You can try to make use of NODE_ENV process env variable from the first task to exclude it for `prod` build

File: src/reducers/index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+export default combineReducers({
+    movies: setMovies,
+    activeMovies: setActiveMovies,
+    genres: setGenres,
+    filter: setFilter,
+    addMovieRequest,
+    editMovieRequest,
+    showModal,
+    movieDetails
```
Comment (lotrien) [inline src/reducers/index.js]: i think this separation doesn't make a lot of sense, it only bring a lot of boilerplate code.
From my perspective, you can have `movies`, `modal`, `filter`, `genres`. You can have as many `cases` in one reducer as you want as far as it bind by its purpose

**Review by lotrien — CHANGES_REQUESTED** [review]

File: src/components/Header/AddMovie.js (hunk 1 above)
Comment (inductio) [reply src/components/Header/AddMovie.js]: Got it, thank you. Fixed.

File: src/components/Main/MoviesGrid.js (hunk 2 above)
Comment (inductio) [reply src/components/Main/MoviesGrid.js]: Cool, thank you. Fixed.

File: src/components/movie/EditMovieForm.js (hunk 4 above)
Comment (inductio) [reply src/components/movie/EditMovieForm.js]: Fixed

File: src/components/movie/MovieCardMenu.js (hunk 5 above)
Comment (inductio) [reply src/components/movie/MovieCardMenu.js]: Fixed. Thank you.

File: src/components/movie/DeleteMovieForm.js (hunk 3 above)
Comment (inductio) [reply src/components/movie/DeleteMovieForm.js]: Fixed

**Review by lotrien — APPROVED** [review]

**Pull request merged**